src/libraries/System.IO.FileSystem.Watcher/tests/Utility/FileSystemWatcherTest.cs

```diff
@@ -374,25 +383,17 @@ public static void ExpectNoError(FileSystemWatcher watcher, Action action, Actio
         /// <param name="action">The Action to execute.</param>
         /// <param name="cleanup">Undoes the action and cleans up the watcher so the test may be run again if necessary.</param>
         /// <param name="attempts">Number of times the test should be executed if it's failing.</param>
-        /// <param name="expected">Whether it is expected that an error event will be arisen.</param>
         /// <returns>True if an Error event was raised by the watcher when the given action was executed; else, false.</returns>
-        public static bool TryErrorEvent(FileSystemWatcher watcher, Action action, Action cleanup, int attempts, bool expected)
+        public static bool TryErrorEvent(FileSystemWatcher watcher, Action action, Action cleanup, int attempts)
         {
             int attemptsCompleted = 0;
-            bool result = !expected;
-            while (result != expected && attemptsCompleted++ < attempts)
+            bool result = false;
+            while (!result && attemptsCompleted++ < attempts)
             {
                 if (attemptsCompleted > 1)
                 {
                     // Re-create the watcher to get a clean iteration.
-                    watcher = new FileSystemWatcher()
-                    {
-                        IncludeSubdirectories = watcher.IncludeSubdirectories,
-                        NotifyFilter = watcher.NotifyFilter,
-                        Filter = watcher.Filter,
-                        Path = watcher.Path,
-                        InternalBufferSize = watcher.InternalBufferSize
-                    };
+                    watcher = RecreateWatcher(watcher);
```

Comment:
Should the recreated watchers be disposed?

Reply:
Ideally, but that's also beyond the scope of your fix.

Let's keep this PR constrained to the minimal changes necessary.